core/src/main/scala/pickling/Macros.scala

```diff
@@ -225,11 +229,16 @@ trait PicklerMacros extends Macro with PickleMacros with FastTypeTagMacros {
           else reflectively("picklee", fir)(fm => putField(q"$fm.get.asInstanceOf[${fir.tpe}]"))
         } else if (fir.javaSetter.isDefined) {
           List(putField(getField(fir)))
-        } else {
+        } else if (fir.isParam) {
           reflectivelyWithoutGetter("picklee", fir)(fvalue =>
             tryPutField(q"$fvalue.asInstanceOf[scala.util.Try[${fir.tpe}]]"))
+        } else {
+          Nil
```

Comment:
> Characterize the difference in Foo and Bar for me in that example? Is it that the reference in Bar.toString causes the scala compiler to generate a field for x0 and in Foo it doesn't generate a field?

I think so.

> But that doesn't seem to match what you're saying about x0 in the example? what am I missing?

The REPL example I posted is from 0.10.x branch. This PR would likely break `Bar` example, but it also prevents `java.lang.Byte` from generating an empty object. I need more code so it works for both cases.